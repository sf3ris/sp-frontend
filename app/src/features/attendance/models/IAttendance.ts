export interface IAttendance {
    _id: string;
    date: string;
    athletes_ids: string[];
    members_ids: string[];
    title: string;
    type?: string;
}
