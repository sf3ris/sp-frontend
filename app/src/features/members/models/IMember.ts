export interface IMember {
    _id : string;
    name : string;
    last_name : string;
    birth_date : string;
    birth_place : string;
    fiscal_code : string;
    address : string;
    zip_code : string;
    city : string;
    province : string;
    gender: string;
    //gender : keyof typeof Gender;
    phone : string;
    email : string;
}

enum Gender {
    M = "M",
    F = "F"
}