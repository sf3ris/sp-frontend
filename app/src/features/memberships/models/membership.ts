export interface IMembership {
    _id: string;
    start_date : Date;
    end_date : Date;
}