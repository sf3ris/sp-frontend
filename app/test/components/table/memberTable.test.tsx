import React from 'react';
import { shallow } from 'enzyme';
import MemberTableComponent from '../../../src/component/member/Table/memberTable.component';
import { IMember } from '../../../src/features/members/models/IMember';

describe('Member table compponent', () => {

    const members : IMember[] = [
        {
            _id : '122',
            name: 'nOME',
            last_name: 'Cognome',
            birth_date: '01/01/1992',
            city: 'Pollenza'
        } as IMember,
        {
            _id : '123',
            name: 'nOME',
            last_name: 'Cognome',
            birth_date: '01/01/1992',
            city: 'Pollenza'
        } as IMember,
        {
            _id : '124',
            name: 'nOME',
            last_name: 'Cognome',
            birth_date: '01/01/1992',
            city: 'Pollenza'
        } as IMember,

    ]

    it('Should match snapshot', () => {

        const wrapper = shallow( <MemberTableComponent 
                members={members} 
                onNew={jest.fn()}
                onDelete={jest.fn()} 
                onEdit={jest.fn()} /> )

        expect(wrapper.debug()).toMatchSnapshot();

    });

    it('Should render 3 rows', () => {

        const wrapper = shallow( <MemberTableComponent 
                members={members} 
                onNew={jest.fn()}
                onDelete={jest.fn()} 
                onEdit={jest.fn()} /> )

        const rows = wrapper.find('MemberRowComponent');

        expect(rows.length).toBe(3);

    });
    
})
