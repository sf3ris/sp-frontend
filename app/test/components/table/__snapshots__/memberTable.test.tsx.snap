// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member table compponent Should match snapshot 1`] = `
"<Fragment>
  <MemberToolbarComponent onPDF={[Function: mockConstructor]} onNew={[Function: mockConstructor]} />
  <WithStyles(ForwardRef(TableContainer))>
    <WithStyles(ForwardRef(Table))>
      <WithStyles(ForwardRef(TableHead))>
        <WithStyles(ForwardRef(TableRow))>
          <WithStyles(ForwardRef(TableCell))>
            Nome
          </WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(TableCell))>
            Cognome
          </WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(TableCell))>
            Data di nascita
          </WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(TableCell))>
            Codice Fiscale
          </WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(TableCell))>
            Citt√†
          </WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(TableCell))>
            Provincia
          </WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(TableCell))>
            Email
          </WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(TableCell))>
            Stato
          </WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(TableCell)) colSpan={2} />
        </WithStyles(ForwardRef(TableRow))>
      </WithStyles(ForwardRef(TableHead))>
      <WithStyles(ForwardRef(TableBody))>
        <MemberRowComponent onDeleteClick={[Function: mockConstructor]} onEditClick={[Function: mockConstructor]} member={{...}} />
        <MemberRowComponent onDeleteClick={[Function: mockConstructor]} onEditClick={[Function: mockConstructor]} member={{...}} />
        <MemberRowComponent onDeleteClick={[Function: mockConstructor]} onEditClick={[Function: mockConstructor]} member={{...}} />
      </WithStyles(ForwardRef(TableBody))>
    </WithStyles(ForwardRef(Table))>
  </WithStyles(ForwardRef(TableContainer))>
</Fragment>"
`;
